tools/plugin_checker/plugin_checker.py

```diff
@@ -0,0 +1,208 @@
+        url = 'https://api.github.com/repos/%s/%s/releases/latest' % (owner, repo)
+        
+        # Create an authenticated session to fetch repo data
+        session = requests.Session()
+        headers = {
+            "Authorization": "token %s" % TOKEN,
+        }
+        r = session.get(url, headers=headers)
+
+        if (r.status_code == 200):
+            repo_data = r.json()
+            # Sometimes the version will be specified in the "name" or "tag_name" fields.
+            # We'll check for both just in case.
+            if compare_versions(v[0], repo_data['name']):
+                PLUGINS_TO_UPDATE = PLUGINS_TO_UPDATE + (k + " needs to be updated to " + repo_data['name']) + "\n"
+                title = k + " needs to be updated to " + repo_data['name']
+                body = "A new version of the " + k + " plugin is available upstream at:\n" + v[1]
+                if args.file:
+                    make_github_issue(title, body, REPO_LABELS)
+            elif compare_versions(v[0], repo_data['tag_name']):
+                PLUGINS_TO_UPDATE = PLUGINS_TO_UPDATE + (k + " needs to be updated to " + repo_data['tag_name']) + "\n"
+                title = k + " needs to be updated to " + repo_data['tag_name']
+                body = "A new version of the " + k + " plugin is available upstream at:\n" + v[1]
+                if args.file:
+                    make_github_issue(title, body, REPO_LABELS)
+            else:
+                PLUGINS_UP_TO_DATE = PLUGINS_UP_TO_DATE + k + "\n"
+        else:
+            PLUGINS_FAILED_TO_CHECK = PLUGINS_FAILED_TO_CHECK + k + "\n"
+    
+    # Update report messages
+    PLUGINS_TO_UPDATE = PLUGINS_TO_UPDATE + "\n"
+    PLUGINS_FAILED_TO_CHECK = PLUGINS_FAILED_TO_CHECK + "\n"
+    PLUGINS_UP_TO_DATE = PLUGINS_UP_TO_DATE + "\n"
+
+def compare_versions(base_version, compared_version):
+    '''Return True if and only if base_version is less than compared_version.'''   
+    # Make sure both inputs aren't None, and aren't empty
+    if base_version and compared_version:
+        if len(base_version) == 0 or len(compared_version) == 0:
+            return False
+        # Strip out any "v" characters in the version strings
+        if base_version[0] == "v":
+            base_version = base_version[1:]
+        if compared_version[0] == "v":
+            compared_version = compared_version[1:]
+        return version.parse(base_version) < version.parse(compared_version)
+    
+    return False
+
+def clean_up():
+    '''Cleans the /tmp directory of the repo that was cloned.'''
+    os.chdir("/tmp")
+    os.system("rm -rf ./%s" % (REPO_NAME))
+
+# Do not run the script if no token is provided
+if not TOKEN:
+    print("No GitHub token provided")
+    sys.exit()
+
+find_plugin_versions()
+check_plugins_for_updates()
+clean_up()
+
+if not args.silent:
+    print("\n" + PLUGINS_TO_UPDATE)
+    print(PLUGINS_UP_TO_DATE)
+    print(PLUGINS_FAILED_TO_CHECK)
```

Comment:
Missing newline

Reply:
At the end of the file or in the print statement?

Reply:
End of the file, if you go into the `files changed` view it shows a little icon saying no newline but for some reason, it doesn't appear here

Reply:
Done!